src/main/java/net/sf/jabref/JabRefGUI.java

```diff
@@ -65,6 +67,13 @@ public JabRefGUI(List<ParserResult> loaded, boolean isBlank) {
         this.loaded = loaded;
         this.isBlank = isBlank;
         openWindow();
+        JabRefGUI.checkVersion(false);
+    }
+
+    public static void checkVersion(boolean manualExecution){
+        Version toBeIgnored = new Version(Globals.prefs.get(JabRefPreferences.VERSION_IGNORED_UPDATE));
+        Version currentVersion = Globals.BUILD_INFO.getVersion();
+        new VersionWorker(JabRefGUI.getMainFrame(), manualExecution, currentVersion, toBeIgnored).execute();
```

Comment:
This means a new `VersionWorker` is created and started each time the method is executed. Here it might not be as impact full because the user probably won't check the version a hundred times manually but you should always cancel a background task before creating the same one again. You can cancel a SwingWorker by creating an object and calling `cancel(true)` on it.